### PR TITLE
CaRT-788: Adding Mercury patch to allow for eager message fix. 

### PR DESCRIPTION
**Pull request by vschhabra**
This is a new HG API to compare internal hg_addr's.

### CODE REVIEW

File: mercury.spec [hunk 1]
```diff
@@ -21,6 +21,7 @@ URL:		http://mercury-hpc.github.io/documentation/
 Source0:	https://github.com/mercury-hpc/%{name}/archive/v%{version}.tar.gz
 Patch1:		https://github.com/mercury-hpc/mercury/compare/v2.0.0a1..%{mercury_commit}.patch
```
Comment (jolivier23) [inline mercury.spec]: does this have any issues?

File: mercury.spec [hunk 2]
```diff
@@ -21,6 +21,7 @@ URL:		http://mercury-hpc.github.io/documentation/
...
 Patch2:		https://github.com/mercury-hpc/mercury/compare/c2c262813811c3ede28ee32fdebbffd417a7cb80..6cbd9d4402db6a34dfb7e9dbf2e826ee90c6a64a.patch
```
Comment (jolivier23) [inline mercury.spec]: Alex just uploaded a patch to remove this patch from the build...should we do that here too?

Comment (vschhabra) [reply mercury.spec]: I will rollback this patch and create a new one

File: mercury.spec (hunk 1 above)
Comment (vschhabra) [reply mercury.spec]: Not sure. I believe is patch 2 which Yulu is rolling back.

**Pull request closed without merging**